packages/seacas/libraries/ioss/src/Ioss_Region.C

```diff
@@ -308,6 +308,56 @@ namespace {
       entity->field_erase(role);
     }
   }
+
+  struct NearestGroupState
+  {
+    std::string group{"/"};
+    int nearestState{-1};
+    double nearestTime{-std::numeric_limits<double>::max()};
+  };
+
+  void get_state_time_round_down(Ioss::DatabaseIO *db, const double targetTime, NearestGroupState& loc)
+  {
+    std::vector<double> timesteps = db->get_db_step_times();
+    int stepCount = timesteps.size();
+
+    double minTimeDiff = loc.nearestState < 0 ? std::numeric_limits<double>::max() : std::abs(loc.nearestTime - targetTime);
+
+    for(int istep = 1; istep <= stepCount; istep++) {
+      double stateTime = timesteps[istep-1];
+      double stepTimeDiff = std::abs(stateTime - targetTime);
+      if(stepTimeDiff <= minTimeDiff) {
+        minTimeDiff = stepTimeDiff;
+        loc.nearestTime = stateTime;
+        loc.nearestState = istep;
+        loc.group = db->get_group_name();
+      }
+    }
+  }
+
+  void locate_state(Ioss::DatabaseIO *db, const double targetTime, NearestGroupState& loc)
+  {
+    // Get state count and all states...
+    std::vector<double> timesteps = db->get_db_step_times();
+    int stepCount = timesteps.size();
+
+    if(targetTime < 0.0) {
+      get_state_time_round_down(db, targetTime, loc);
+    }
+    else {
+      double minTimeDiff = loc.nearestState < 0 ? std::numeric_limits<double>::max() : std::fabs(targetTime - loc.nearestTime);
+      for(int istep = 1; istep <= stepCount; istep++) {
+        double stateTime = timesteps[istep-1];
+        double stepTimeDiff = std::fabs(targetTime - stateTime);
```

Comment:
Is there a reason why `std::fabs` is used here, but `std::abs` is used on line 328 above.

Reply:
None ... it is what is being used in Frio_Restart so wasn't sure if the distinction was needed. Should I make them consistent?

Reply:
Yes, probably.

Reply:
Done